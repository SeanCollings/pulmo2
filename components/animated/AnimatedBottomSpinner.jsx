import React, { useState, useEffect } from 'react';
import { StyleSheet, Animated, ActivityIndicator, View } from 'react-native';
import { useTheme } from '../../hooks/useTheme';

const AnimatedBottomSpinner = ({
  animateInFinished,
  animateOutFinished,
  animateIn,
  animateOut,
  duration = 200,
}) => {
  const theme = useTheme();
  const [animatedHeight] = useState(new Animated.Value(100));

  useEffect(() => {
    if (animateIn) {
      Animated.timing(animatedHeight, {
        toValue: 0,
        duration,
        useNativeDriver: true,
      }).start(({ finished }) => {
        if (finished) {
          animateInFinished();
        }
      });
    }
  }, [animateIn]);

  useEffect(() => {
    if (animateOut) {
      Animated.timing(animatedHeight, {
        toValue: 100,
        duration,
        useNativeDriver: true,
      }).start(({ finished }) => {
        if (finished) {
          animateOutFinished();
        }
      });
    }
  }, [animateOut]);

  const animatedStyleHeight = {
    translateY: animatedHeight,
  };

  return (
    <View style={styles.spinnerContainer}>
      <Animated.View
        style={{
          ...styles.spinnerMore,
          ...animatedStyleHeight,
          backgroundColor: theme.BACKGROUND,
          borderColor: theme.DARK ? theme.PRIMARY : theme.BORDER,
        }}
      >
        <ActivityIndicator size="large" color={theme.SECONDARY} />
      </Animated.View>
    </View>
  );
};

AnimatedBottomSpinner.defaultProps = {
  animateInFinished: () => {},
  animateOutFinished: () => {},
  animateIn: false,
  animateOut: false,
};

const styles = StyleSheet.create({
  spinnerContainer: {
    position: 'absolute',
    alignSelf: 'center',
    bottom: 0,
  },
  spinnerMore: {
    padding: 10,
    borderTopLeftRadius: 50,
    borderTopRightRadius: 50,
    borderWidth: 1,
    borderBottomWidth: 0,
  },
});

export default AnimatedBottomSpinner;
