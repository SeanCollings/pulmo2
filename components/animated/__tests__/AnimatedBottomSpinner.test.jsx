import React from 'react';
import { act, create } from 'react-test-renderer';
import { cleanup } from 'react-native-testing-library';
import AnimatedBottomSpinner from '../AnimatedBottomSpinner';

jest.useFakeTimers();

describe('AnimatedAppear - unit test', () => {
  let component;
  afterEach(() => {
    cleanup();
  });

  test('should render default', async () => {
    await act(async () => {
      component = create(<AnimatedBottomSpinner duration={0} />);
    });

    expect(component).toMatchSnapshot();
  });

  test('should render with animateIn', async () => {
    await act(async () => {
      component = create(<AnimatedBottomSpinner animateIn duration={0} />);
    });

    expect(component).toMatchSnapshot();
  });

  test('should render with animateOut', async () => {
    await act(async () => {
      component = create(<AnimatedBottomSpinner animateOut duration={0} />);
    });

    expect(component).toMatchSnapshot();
  });
});
