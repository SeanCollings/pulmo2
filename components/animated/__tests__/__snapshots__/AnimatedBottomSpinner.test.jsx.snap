// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedAppear - unit test should render default 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "bottom": 0,
      "position": "absolute",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderBottomWidth": 0,
        "borderColor": "#CCC",
        "borderTopLeftRadius": 50,
        "borderTopRightRadius": 50,
        "borderWidth": 1,
        "padding": 10,
        "translateY": 100,
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#002f56"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
</View>
`;

exports[`AnimatedAppear - unit test should render with animateIn 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "bottom": 0,
      "position": "absolute",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderBottomWidth": 0,
        "borderColor": "#CCC",
        "borderTopLeftRadius": 50,
        "borderTopRightRadius": 50,
        "borderWidth": 1,
        "padding": 10,
        "translateY": 0,
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#002f56"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
</View>
`;

exports[`AnimatedAppear - unit test should render with animateOut 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "bottom": 0,
      "position": "absolute",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderBottomWidth": 0,
        "borderColor": "#CCC",
        "borderTopLeftRadius": 50,
        "borderTopRightRadius": 50,
        "borderWidth": 1,
        "padding": 10,
        "translateY": 100,
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#002f56"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
</View>
`;
